src/plugins/ui_feature/cell_computed_style.ts

```diff
@@ -0,0 +1,101 @@
+import { LINK_COLOR } from "../../constants";
+import { isObjectEmptyRecursive, removeFalsyAttributes } from "../../helpers/index";
+import {
+  Command,
+  doesCommandInvalidatesTableStyle,
+  invalidateCFEvaluationCommands,
+  invalidateEvaluationCommands,
+} from "../../types";
+import { Border, CellPosition, Style, UID } from "../../types/misc";
+import { UIPlugin } from "../ui_plugin";
+
+export class CellComputedStylePlugin extends UIPlugin {
+  static getters = ["getCellComputedBorder", "getCellComputedStyle"] as const;
+
+  private styles: Record<UID, Record<number, Record<number, Style>>> = {};
+  private borders: Record<UID, Record<number, Record<number, Border | null>>> = {};
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      cmd.type === "UPDATE_CELL" ||
+      cmd.type === "EVALUATE_CELLS"
+    ) {
+      this.styles = {};
+      this.borders = {};
+      return;
+    }
+
+    if (doesCommandInvalidatesTableStyle(cmd)) {
+      delete this.styles[cmd.sheetId];
+      delete this.borders[cmd.sheetId];
+      return;
+    }
+
+    if (invalidateCFEvaluationCommands.has(cmd.type)) {
+      this.styles = {};
+      return;
+    }
+  }
+
+  getCellComputedBorder(position: CellPosition): Border | null {
+    const { sheetId, row, col } = position;
+    if (this.borders[sheetId]?.[row]?.[col] !== undefined) {
+      return this.borders[sheetId][row][col];
+    }
+    if (!this.borders[sheetId]) {
+      this.borders[sheetId] = {};
+    }
+    if (!this.borders[sheetId][row]) {
+      this.borders[sheetId][row] = {};
+    }
+    if (!this.borders[sheetId][row][col]) {
+      this.borders[sheetId][row][col] = this._getCellComputedBorder(position);
+    }
+    return this.borders[sheetId][row][col];
+  }
+
+  getCellComputedStyle(position: CellPosition): Style {
+    const { sheetId, row, col } = position;
+    if (this.styles[sheetId]?.[row]?.[col] !== undefined) {
+      return this.styles[sheetId][row][col];
+    }
+    if (!this.styles[sheetId]) {
+      this.styles[sheetId] = {};
+    }
+    if (!this.styles[sheetId][row]) {
+      this.styles[sheetId][row] = {};
+    }
+    if (!this.styles[sheetId][row][col]) {
+      this.styles[sheetId][row][col] = this._getCellComputedStyle(position);
+    }
+    return this.styles[sheetId][row][col];
+  }
+
+  private _getCellComputedBorder(position: CellPosition): Border | null {
+    const cellBorder = this.getters.getCellBorder(position) || {};
+    const cellTableBorder = this.getters.getCellTableBorder(position) || {};
+
+    // Use removeFalsyAttributes to avoid overwriting borders with undefined values
+    const border = { ...cellTableBorder, ...removeFalsyAttributes(cellBorder) };
```

Comment:
Maybe add a removeFalsyAttributes for `cellTableBorder` to avoid having undefined values in the final border object (just as you did it for the computedStyle below) ?